### PR TITLE
Make MAC tag length variable

### PR DESCRIPTION
**Pull request by claucece**
This is related to #205 . It is not perfect by any means, but I was thinking it could be an starting point.
Comments welcome @raphaelrobert @franziskuskiefer

### CODE REVIEW

**Review by kkohbrok — DISMISSED** [review]
Thanks for the PR!

Looks good to me! At some point, the mac length should probably be exposed via `AeadMode`, but I think `evercrypt` just hardcodes it to 16 at the moment, so this is a good step in the right direction.

File: src/ciphersuite/mod.rs [hunk 1]
```diff
@@ -562,6 +570,7 @@ impl AeadKey {
         AeadKey {
             aead_mode,
             value: aead_key_gen(aead_mode),
+            mac_len: 16,
```
Comment (raphaelrobert) [inline src/ciphersuite/mod.rs:573]: We could pass `&Ciphersuite` as an argument here as well in order to avoid the hard-coding

**Review by raphaelrobert — CHANGES_REQUESTED** [review]
Thanks for the PR! LGTM with one nit

**Comment by franziskuskiefer** [comment]
> Looks good to me! At some point, the mac length should probably be exposed via `AeadMode`, but I think `evercrypt` just hardcodes it to 16 at the moment, so this is a good step in the right direction.

yeah this is currently hard coded. But I'm happy to add something. I want to make a new release soon anyway. File an issue or a PR over there https://www.franziskuskiefer.de/evercrypt-rust

**Comment by kkohbrok** [comment]
> > Looks good to me! At some point, the mac length should probably be exposed via `AeadMode`, but I think `evercrypt` just hardcodes it to 16 at the moment, so this is a good step in the right direction.
> 
> yeah this is currently hard coded. But I'm happy to add something. I want to make a new release soon anyway. File an issue or a PR over there https://github.com/franziskuskiefer/evercrypt-rust/

I've created a PR with a relatively primitive solution that would allow this over in the evercrypt repo.

File: src/ciphersuite/mod.rs (hunk 1 above)
Comment (claucece) [reply src/ciphersuite/mod.rs]: Just added it ;)

**Comment by claucece** [comment]
Oh, thank you @kkohbrok should we add that as a new PR once a new release of evercrypt happens?

**Comment by raphaelrobert** [comment]
> Oh, thank you @kkohbrok should we add that as a new PR once a new release of evercrypt happens?

I think @franziskuskiefer was fast for that one: https://github.com/franziskuskiefer/evercrypt-rust/pull/19
But I suppose it requires a new release of evercrypt?

**Comment by franziskuskiefer** [comment]
> But I suppose it requires a new release of evercrypt?

Yeah this needs a new release (or we have to point to the git version in the meantime). I'm waiting for the new HPKE test vectors before I do the new releases. This probably takes a couple more days.

**Comment by raphaelrobert** [comment]
Then we can merge this and do a follow-up PR.

**Review by raphaelrobert — DISMISSED** [review]

**Review by raphaelrobert — APPROVED** [review]

**Review by kkohbrok — APPROVED** [review]

**Pull request merged**